app/models/idea.rb

```diff
@@ -1,7 +1,18 @@
 class Idea < ApplicationRecord
   belongs_to :user
   validates :title, presence: true
-  
+  validates :area_of_interest, presence: true, if: :submitted?
+  validates :business_area, presence: true, if: :submitted?
+  validates :it_system, presence: true, if: :submitted?
+  validates :idea, presence: true, if: :submitted?
+  validates :benefits, presence: true, if: :submitted?
+  validates :impact, presence: true, if: :submitted?
+  validates :involvement, presence: true, if: :submitted?
+
+  def submitted?
+    submission_date
```

Comment:
I'm a bit confused here, this works but :submission_date doesn't. I used the docs here - https://guides.rubyonrails.org/active_record_validations.html#using-a-symbol-with-if-and-unless
I understand what the method is doing, but not how, where is submission_date coming from here?  I'm assuming it's some validates magic but cant find it in the docs.

Reply:
not sure why :submission_date doesn't work out of the box, but in any case line 13 would be better as `submission_date.present?` as it's a boolean